AWSAuthSDK/Sources/AWSMobileClient/AWSMobileClientExtensions.swift

```diff
@@ -664,6 +664,30 @@ extension AWSMobileClient {
         self.userPoolClient?.clearAll()
     }
     
+    /// Asynchronous deleteUser method which requires network activity.
+    ///
+    /// - Parameters:
+    ///   - completionHandler: completion handler for success or error callback.
+    public func deleteUser(signOut: Bool = true, completionHandler: @escaping ((Error?) -> Void)) {
```

Comment:
nit: I see force unwrapping done here and elsewhere so it appears to be common with this extension. It would be better to guard and call `fatalError` with a useful message if this value is ever nil.

Reply:
This cannot be nil if there is a valid CongnitoUserPool configuration.  If there's not a valid configuration, an error will be thrown during configuration, so we will never get to this point.  With that said, I agree that it's better to avoid force unwrapping since the effects are not obvious when reading the code.  I changed this to handle it explicitly.  As you said, this is now handled differently than the rest of this extension, but I feel this is preferable, and this could serve as an example to update the other methods going forward.

Reply:
👍